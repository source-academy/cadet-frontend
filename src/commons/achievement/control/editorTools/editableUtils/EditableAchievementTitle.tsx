import { EditableText } from '@blueprintjs/core';

type EditableAchievementTitleProps = {
  title: string;
  changeTitle: any;
};

function EditableAchievementTitle(props: EditableAchievementTitleProps) {
  const { title, changeTitle } = props;

  return (
    <div>
      <h1>
        <EditableText placeholder={`Enter your title here`} value={title} onChange={changeTitle} />
      </h1>
    </div>
  );
}

export default EditableAchievementTitle;
