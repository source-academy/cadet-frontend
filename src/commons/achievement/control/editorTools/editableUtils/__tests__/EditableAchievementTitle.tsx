import { mount } from 'enzyme';
import React from 'react';

import EditableAchievementTitle from '../EditableAchievementTitle';

const mockProps = {
  title: '',
  changeTitle: () => {}
};

test('EditableAchievementTitle component renders correctly', () => {
  const goal = <EditableAchievementTitle {...mockProps} />;
  const tree = mount(goal);
  expect(tree.debug()).toMatchSnapshot();
});
