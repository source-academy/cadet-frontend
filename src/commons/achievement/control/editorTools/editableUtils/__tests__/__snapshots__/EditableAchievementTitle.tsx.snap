// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableAchievementTitle component renders correctly 1`] = `
"<EditableAchievementTitle title=\\"\\" changeTitle={[Function: changeTitle]}>
  <div>
    <h1>
      <Blueprint3.EditableText placeholder=\\"Enter your title here\\" value=\\"\\" onChange={[Function: changeTitle]} alwaysRenderInput={false} confirmOnEnterKey={false} defaultValue=\\"\\" disabled={false} maxLines={Infinity} minLines={1} minWidth={80} multiline={false} type=\\"text\\">
        <div className=\\"bp3-editable-text bp3-editable-text-placeholder\\" onFocus={[Function (anonymous)]} tabIndex={0}>
          <span className=\\"bp3-editable-text-content\\" style={{...}}>
            Enter your title here
          </span>
        </div>
      </Blueprint3.EditableText>
    </h1>
  </div>
</EditableAchievementTitle>"
`;
